### PR TITLE
Lv 160

### PR DESCRIPTION
**Pull request by TamaraKovalchuk**
Moved function about user to user.js from routes/index.js and get user by ID in profile

### CODE REVIEW

File: models/user.js [hunk 1]
```diff
@@ -142,32 +146,65 @@ UserSchema.statics.authenticateSocial = async (req, res, next) => {
...
+      {
+        $set: {
+          userName: req.body.userName,
+          nickname: req.body.nickname,
+          website: req.body.website,
+          userBio: req.body.userBio
+        }
+      }, function (err, userProfile) {
+        if (err) return next(err);
+        res.status(200).send({ userprofile: prepareUser(userProfile) });
```
Comment (Jestermaxrko) [inline models/user.js]: use json instead of send

File: models/user.js [hunk 2]
```diff
@@ -142,32 +146,65 @@ UserSchema.statics.authenticateSocial = async (req, res, next) => {
...
+      });
+  } else {
+    res.status(404).send();
```
Comment (Jestermaxrko) [inline models/user.js]: you need create error object with status 404 and use next(err)

Comment (TamaraKovalchuk) [reply models/user.js]: let error = new Error ('Eror msg')
    error.status = 404;
    next(error);

File: models/user.js (hunk 1 above)
Comment (TamaraKovalchuk) [reply models/user.js]: res.status(200).json({ userprofile: prepareUser(userProfile) });

File: routes/index.js [hunk 3]
```diff
@@ -131,42 +129,8 @@ router.post("/search", async (req, res) => {
...
-  });
-  if (profile.length > 0)
-  { 
-    
-    res.status(200).json({userprofile: prepareUser(profile[0])});
-  } else {
-    res.status(404).send();
-  }
-});
+router.get('/profile/:nickname', User.userProfile);
```
Comment (carbonid1) [inline routes/index.js]: make a separate file for the profile routes

**Review by Dvoryan — APPROVED** [review]

**Review by carbonid1 — APPROVED** [review]

**Pull request merged**